src/pages/index.js

```diff
@@ -0,0 +1,273 @@
+        finally(() => {
+            clearInterval(loadingAnimation);
+            getSubmitButton(popup).textContent = originalText;
+            editAvatarForm.reset();
+            disableButton(editAvatarForm);
+            togglePopup(popup);
+        });
+    });
+}
+
+function renderCard(data) {
+    cardList.prepend(getCardElement(data));
+}
+
+function openFullImage(data) {
+    fullImageSelectors.fullImageImgElement.src = "";
+    fullImageSelectors.fullImageImgElement.alt = "";
+    fullImageSelectors.fullImageText.textContent = "";
+    togglePopup(fullImageSelectors.fullImagePopup);
+    fullImageSelectors.fullImageImgElement.src = data.src;
+    fullImageSelectors.fullImageImgElement.alt = data.alt;
+    fullImageSelectors.fullImageText.textContent = data.name;
+}
+
+function setDeleteCardPopupListeners(card) {
+    const deleteButton = deleteCardSelectors.deleteCardButtonDelete;
+    const cancelButton = deleteCardSelectors.deleteCardButtonCancel;
+
+    const deleteButtonHandler = () => {
+        return function handler() {
+            const {loadingAnimation, originalText} = submitButtonText(deleteCardSelectors.deleteCardPopup);
+            api.deleteCard(card.id).catch(err => console.log(err)).finally(() => {
+                clearInterval(loadingAnimation);
+                getSubmitButton(deleteCardSelectors.deleteCardPopup).textContent = originalText;
+                togglePopup(deleteCardSelectors.deleteCardPopup);
+                card.remove();
+                deleteButton.removeEventListener("click", handler);
+            });
+        };
+    };
+
+    const cancelButtonHandler = () => {
+        togglePopup(deleteCardSelectors.deleteCardPopup);
+        cancelButton.removeEventListener("click", cancelButtonHandler);
+    };
+
+    return () => {
+        deleteButton.removeEventListener("click", deleteButton._handler);
+        deleteButton._handler = deleteButtonHandler(card);
+        deleteButton.addEventListener("click", deleteButton._handler);
+
+        cancelButton.removeEventListener("click", cancelButtonHandler);
+        cancelButton.addEventListener("click", cancelButtonHandler);
+    };
+}
+
+function getCardElement(data) {
+    const cardElement = document
+        .querySelector("#card-template")
+        .content.querySelector(".photos__list-item")
+        .cloneNode(true);
+    cardElement.querySelector(".card__text").textContent = data.name;
+    const cardPhoto = cardElement.querySelector(".card__image");
+    const cardElementLike = cardElement.querySelector(".card__like-button");
+    cardPhoto.src = data.src;
+    cardPhoto.alt = data.alt;
+    cardElement.id = data.id;
+    if (data.isLiked){
+        cardElementLike.classList.toggle("card__like-button_active");
+    }
+
+    cardElementLike.addEventListener("click", () => {
+        api.likeCard(data.id).catch(err => console.log(err));
+        cardElementLike.classList.toggle("card__like-button_active");
```

Comment:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.